.github/workflows/generate-pickle-files.yml

```diff
@@ -0,0 +1,46 @@
+name: Generate static pickle files
+
+on: pull_request
```

Comment:
Indeed, I think we should not run this at each PR. Let's use a manual trigger instead: https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow